app/src/main/java/space/taran/arknavigator/mvp/model/repo/preview/GeneralPreviewGenerator.kt

```diff
@@ -0,0 +1,50 @@
+package space.taran.arknavigator.mvp.model.repo.preview
+
+import android.util.Log
+import space.taran.arknavigator.mvp.model.repo.preview.generator.ImagePreviewGenerator
+import space.taran.arknavigator.mvp.model.repo.preview.generator.LinkPreviewGenerator
+import space.taran.arknavigator.mvp.model.repo.preview.generator.PdfPreviewGenerator
+import space.taran.arknavigator.mvp.model.repo.preview.generator.PreviewGenerator
+import space.taran.arknavigator.mvp.model.repo.preview.generator.TxtPreviewGenerator
+import space.taran.arknavigator.utils.LogTags.PREVIEWS
+import space.taran.arknavigator.utils.extension
+import space.taran.arknavigator.utils.getMimeTypeUsingTika
+import java.nio.file.Path
+import kotlin.system.measureTimeMillis
+
+object GeneralPreviewGenerator {
+
+    // Use this list to declare new types of generators
+    private val generators = listOf(
+        ImagePreviewGenerator,
+        LinkPreviewGenerator,
+        PdfPreviewGenerator,
+        TxtPreviewGenerator,
+    )
+
+    fun generate(path: Path, previewPath: Path, thumbnailPath: Path) {
+        val generator = findGenerator(path) ?: let {
+            Log.d(PREVIEWS, "No generators found for $path")
+            return
+        }
+        val time = measureTimeMillis {
+            generator.generate(path, previewPath, thumbnailPath)
+        }
+        Log.d(PREVIEWS, "Preview and thumbnail generated for $path in $time ms")
+    }
+
+    private fun findGenerator(path: Path): PreviewGenerator? {
+        var generator = generators.find { it.isValid(path) }
+        if (generator != null) return generator
```

Comment:
Aha.. So we are calling Tika only if we haven't found any generator for the resource basing on its path. That's good, no need to worry about performance (I think reading first 100K characters is a bit demanding).